README.md

```diff
@@ -1,50 +1,204 @@
-  dependency requirements
+- how to easily integrate a library written in Fortran with a library written in Python
+- how to easily integrate Python libraries that are likely to have incompatible dependency requirements
 
-We want to solve this problems in such a way that users need no knowledge of
-messaging queues, serialization techniques, or remote-procedure-protocols.
+We want to solve this problems in such a way that users need no knowledge of messaging queues, serialization techniques, or remote-procedure-protocols.
 
-Inspiration was drawn from the client-kernel model of the IPykernel/Jupyter
-ecosystem, RPC packages such as PyRPC, and Dask's task submission patterns.
+Approach
+---------
 
-# Examples
+`irpc` follows a client/kernel model where the client and kernel define a set of endpoints through which the user submits data through the client to the kernel and the kernel responds with the results from the function corresponding to that endpoint. Lucas has extended this approach by implementing a class proxy protocol on top of it which allows for the sequential management of state on the kernel side.
 
-- C- Test via*:
+"kernel" is defined as a process that executes code upon request. A "client" facilitates the interaction between the user who is in one environment (and process) and a "kernel" which is in a different process and potentially different environment.
+
+The client and kernel do not need to be in the same language or computing environment. So far, this project has successfully integrated Python and Fortran libraries as well as Python libraries in different conda environments. The upside of this approach is that you could simultaneously run kernels in R, C, Fortran (with C-bindings), and Python. The user could spin up a client for each kernel and submit data to them. The kernels could even spin up clients to communicate with kernels written in other languages. Note that the same kernel can accept data from clients in multiple languages. That is, special kernels for the Python-to-R interaction and another for the Python-to-Fortran interaction are not needed. Only a Python kernel with the correct endpoints is needed.
+
+API
+----
+
+I reference the Python API for this project because it closer to our vision for a more stable API.
+
+Key Objects:
+- `Kernel` -- registers handlers and executes tasks corresponding to them
+- `Client` -- submits tasks and arguments to kernel and returns result; handles data serialization and port registration
+- `KernelManager` -- spins up `Kernel` instances in subprocesses
+
+Key usage pattern:
 
-Build and Start C kernel:
 ```
-docker build -t taxsimlink ./
-docker run -p 5566:5566 -p 5567:5567 -t taxsimlink ./linked
+from rpc.kernelmanager import KernelManager
+from rpc.client import Client
+
+kernel_info = {
+    'taxcalc1': {
+        'module_path': 'taxcalc_kernel.py',
+        'executable': '/Users/henrydoupe/anaconda3/envs/taxcalc-dev/bin/python'
```

Comment:
Is it possible to just specify `python` and have the script look for it in the current `PATH` (e.g. the environment)?

Reply:
The `'executable'` key is only necessary when you want to use a different conda environments. Useful abstractions could be:
- specifying the conda env name and figuring out the path using `sys.executable`
- allow users to specify conda envs, virtual envs, pipenvs, etc.

Reply:
Thanks for the clarification. I agree that that would be helpful.